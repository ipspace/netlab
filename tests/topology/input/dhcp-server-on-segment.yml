module: [dhcp]

nodes:
  h1:
    device: linux
  dhs:
    device: dnsmasq
    provider: clab

links:
- h1:
  dhs:
  dhcp.client.ipv4: true
