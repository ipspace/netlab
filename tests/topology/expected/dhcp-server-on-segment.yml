dhcp:
  pools:
  - clean_name: links_1_
    excluded:
      ipv4:
      - 172.16.0.2
    ipv4: 172.16.0.0/24
    name: links[1]
input:
- topology/input/dhcp-server-on-segment.yml
- package:topology-defaults.yml
libvirt:
  providers:
    clab: true
links:
- _linkname: links[1]
  bridge: input_1
  dhcp:
    client:
      ipv4: true
    subnet:
      ipv4: true
  interfaces:
  - dhcp:
      client:
        ipv4: true
    ifindex: 1
    ifname: eth1
    ipv4: 172.16.0.1/24
    node: h1
  - dhcp:
      client:
        ipv4: true
    ifindex: 1
    ifname: eth1
    ipv4: 172.16.0.2/24
    node: dhs
  libvirt:
    provider:
      clab: true
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 172.16.0.0/24
  role: stub
  type: lan
module:
- dhcp
name: input
nodes:
  dhs:
    _daemon: true
    _daemon_config:
      dhcp: /etc/dhcp.ignore
      dnsmasq: /etc/dnsmasq.conf
    _daemon_parent: linux
    af:
      ipv4: true
    box: netlab/dnsmasq:latest
    clab:
      binds:
      - clab_files/dhs/hosts:/etc/hosts
      - clab_files/dhs/dnsmasq:/etc/dnsmasq.conf
      - clab_files/dhs/dhcp:/etc/dhcp.ignore
      config_templates:
      - hosts:/etc/hosts
      - dnsmasq:/etc/dnsmasq.conf
      - dhcp:/etc/dhcp.ignore
      kind: linux
    device: dnsmasq
    dhcp:
      pools:
      - clean_name: links_1_
        excluded:
          ipv4:
          - 172.16.0.2
        ipv4: 172.16.0.0/24
        name: links[1]
      server: true
    hostname: clab-input-dhs
    id: 2
    interfaces:
    - bridge: input_1
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.0.2/24
      linkindex: 1
      mtu: 1500
      name: dhs -> h1
      neighbors:
      - dhcp:
          client:
            ipv4: true
        ifname: eth1
        ipv4: 172.16.0.1/24
        node: h1
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    module:
    - dhcp
    mtu: 1500
    name: dhs
    provider: clab
    role: host
  h1:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 1
    interfaces:
    - bridge: input_1
      dhcp:
        client:
          ipv4: true
      ifindex: 1
      ifname: eth1
      linkindex: 1
      name: h1 -> dhs
      neighbors:
      - dhcp:
          client:
            ipv4: true
        ifname: eth1
        ipv4: 172.16.0.2/24
        node: dhs
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    module:
    - dhcp
    name: h1
    role: host
provider: libvirt
